lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,363 @@
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        results = dict()
+        changed = False
+        image = None
+
+        if not self.location:
+            # Set default location
+            resource_group = self.get_resource_group(self.resource_group)
+            self.location = resource_group.location
+
+        self.log('Fetching image {0}'.format(self.name))
+        image = self.get_image()
+        if image:
+            self.check_provisioning_state(image, self.state)
+            results = self.image_to_dict(image)
+            # update is not supported
+            if self.state == 'absent':
+                changed = True
+        # the image does not exist and create a new one
+        elif self.state == 'present':
+            changed = True
+
+        self.results['changed'] = changed
+        self.results['state'] = results
+
+        if not self.check_mode and changed:
+
+            if self.state == 'present':
+                # create from virtual machine
+                vm_resource_id = format_resource_id(self.source,
+                                                    self.subscription_id,
+                                                    'Microsoft.Compute',
+                                                    'virtualMachines',
+                                                    self.resource_group)
+                resource = parse_resource_id(vm_resource_id)
+                vm = self.get_vm(resource['resource_group'], resource['name']) if resource['type'] == 'virtualMachines' else None
+                if vm and self.data_disk_sources:
+                    self.fail('data_disk_sources is not allowed when capturing image from vm')
+
+                if vm:
+                    os_type = vm.storage_profile.os_disk.os_type.value
+                    image = Image(self.location, source_virtual_machine=SubResource(vm_resource_id))
+                else:
+                    if not self.os_type:
+                        self.fail('os_type is required to create the image')
+                    os_disk = self.create_os_disk()
+                    data_disks = self.create_data_disks()
+                    storage_profile = ImageStorageProfile(os_disk=os_disk, data_disks=data_disks)
+                    image = Image(self.location, storage_profile=storage_profile, tags=self.tags)
+                new_image = self.create_image(image)
+                self.results['state'] = self.image_to_dict(new_image)
+
+            elif self.state == 'absent':
+                # delete image
+                self.delete_image()
+                # the delete does not actually return anything. if no exception, then we'll assume
+                # it worked.
+                self.results['state']['status'] = 'Deleted'
+
+        return self.results
+
+    def image_to_dict(self, image):

```

Comment:
`image_to_dict` -> `_image_to_dict` to make it protected, if it is not supposed to be used by other modules.